openlibrary/core/vendors.py

```diff
@@ -43,92 +44,83 @@ def search_amazon(title='', author=''):
-        'price': price_fmt,
-        'price_amt': price,
-        'qlt': qlt,
-        'title': product.title,
-        'authors': [{'name': name} for name in product.authors],
         'source_records': ['amazon:%s' % product.asin],
-        'number_of_pages': product.pages,
-        'languages': list(product.languages),
-        'cover': product.large_image_url,
-        'product_group': product.product_group,
+        'isbn_10': [product.asin],
+        'isbn_13': [isbn_10_to_isbn_13(product.asin)],
+        'price': price and price.display,
+        'price_amt': price.value and int(100 * price.value),
+        'title': product.title,
+        'cover': product.images and product.images.large,
+        'authors': [{'name': contrib.name, 'role': contrib.role}
+                    for contrib in attribution.contributors],
+        'publishers': attribution.brand and [attribution.brand.display_value],
+        'number_of_pages': (edition_info.pages_count and
+                            edition_info.pages_count.display_value),
+        'edition_num': (edition_info.edition and
+                        edition_info.edition.display_value),
+        'publish_date': (edition_info.publication_date and
+                             edition_info.publication_date.display_value),
+        'languages': (
+            edition_info.languages and
+            list(set(lang.display_value
+                     for lang in edition_info.languages.display_values))
+        ),
+        'physical_format': (
+            item_info.classifications and
+            getattr(item_info.classifications.binding, 'display_value')),
+        'dimensions': dims and {
+            d: [getattr(dims, d).display_value, getattr(dims, d).unit]
+            for d in dims.to_dict()}
     }
-    if product._safe_get_element('OfferSummary') is not None:
-        data['offer_summary'] = {
-            'total_new': int(product._safe_get_element_text('OfferSummary.TotalNew')),
-            'total_used': int(product._safe_get_element_text('OfferSummary.TotalUsed')),
-            'total_collectible': int(product._safe_get_element_text('OfferSummary.TotalCollectible')),
-        }
-        collectible = product._safe_get_element_text('OfferSummary.LowestCollectiblePrice.Amount')
-        if new:
-            data['offer_summary']['lowest_new'] = int(new)
-        if used:
-            data['offer_summary']['lowest_used'] = int(used)
-        if collectible:
-            data['offer_summary']['lowest_collectible'] = int(collectible)
-        amazon_offers = product._safe_get_element_text('Offers.TotalOffers')
-        if amazon_offers:
-            data['offer_summary']['amazon_offers'] = int(amazon_offers)
-
-    if product.publication_date:
-        data['publish_date'] = product._safe_get_element_text('ItemAttributes.PublicationDate')
-        if re.match(AMAZON_FULL_DATE_RE, data['publish_date']):
-            data['publish_date'] = product.publication_date.strftime('%b %d, %Y')
```

Comment:
The new date doesn't look like it is being formatted from the example you posted, and this removed code.

Reply:
fixed date formatting